Assets/Scripts/Puzzle.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+/// <summary>
+/// A puzzle is an object connected to a switch or group of switches.
+/// It has a solution, a string formed from a combination of each switch's character.
+/// When the player activates the switches in a sequence which matches this solution,
+/// the puzzle's state changes to allow the player to progress.
+/// </summary>
+public abstract class Puzzle : MonoBehaviour
+{
+    [SerializeField]
+    [Tooltip("The solution to the puzzle.\n This should be a combination " +
+        "of the connected switches' solutionEntries.")]
+    private string solution = "1";
+
```

Comment:
Does a puzzle need to know about it's connected switches? If it did, you could probably assert if the solution is possible from the connected switches characters and throw an exception if it's not.

Reply:
I'll make an issue of that and consider implementing it later. Unless you strongly disagree, I'm not sure it's crucial enough to work on right at this moment. I do like the idea, though.